### PR TITLE
Handle variable copy/paste shortcut in Clipboard Widget

### PR DESCRIPTION
**Pull request by marmarta**
Show correct copy/paste shortcut in the widget and notifications, update it as needed when corresponding feature changes.

fixes QubesOS/qubes-issues#7777

### CODE REVIEW

File: qui/clipboard.py [hunk 1]
```diff
@@ -259,15 +283,59 @@ def send_notify(self, body):
...
+    for d in done:  # pylint: disable=invalid-name
+        try:
+            d.result()
+        except Exception:  # pylint: disable=broad-except
+            exc_type, exc_value = sys.exc_info()[:2]
+            dialog = Gtk.MessageDialog(
+                None, 0, Gtk.MessageType.ERROR, Gtk.ButtonsType.OK)
+            dialog.set_title(_("Houston, we have a problem..."))
+            dialog.set_markup(_(
+                "<b>Whoops. A critical error in Qubes Update has occurred.</b>"
```
Comment (marmarek) [inline qui/clipboard.py:328]: Too much copy&paste.

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarta** [comment]
PipelineRetry

File: qui/clipboard.py [hunk 2]
```diff
@@ -79,10 +85,10 @@ def _copy(self, vmname: str = None):
...
         size = clipboard_formatted_size()
 
-        body = _("Clipboard contents fetched from qube: <b>'{0}'</b>\n"
-                 "Copied <b>{1}</b> to the global clipboard.\n"
-                 "<small>Press Ctrl+Shift+V in qube to paste to local"
-                 "clipboard.</small>").format(vmname, size)
+        body = _(f"Clipboard contents fetched from qube: <b>'{vmname}'</b>\n"
+                 f"Copied <b>{size}</b> to the global clipboard.\n"
+                 f"<small>Press {self.gtk_app.paste_shortcut} in qube "
+                 "to paste to local clipboard.</small>")
```
Comment (marmarek) [inline qui/clipboard.py:91]: `f""`applied this way isn't compatible with gettext. Please leave it as a `format()` call for now (or similar alternative).

File: qui/utils.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        except Exception as _ex:  # pylint: disable=broad-except
+            exc_type, exc_value = sys.exc_info()[:2]
+            dialog = Gtk.MessageDialog(
+                None, 0, Gtk.MessageType.ERROR, Gtk.ButtonsType.OK)
+            dialog.set_title(_("Houston, we have a problem..."))
+            dialog.set_markup(message)
+            exc_description = "\n<b>{}</b>: {}\n{}".format(
+                   exc_type.__name__, exc_value, traceback.format_exc(limit=10)
+                )
+            dialog.format_secondary_markup(escape(exc_description))
```
Comment (marmarek) [inline qui/utils.py:68]: The use of `escape()` here looks suspicious, will break `<b>` tags above. I think you need to escape each format's argument separately.

File: qui/clipboard.py (hunk 2 above)
Comment (marmarta) [reply qui/clipboard.py:91]: fixed

**Comment by marmarta** [comment]
Fixed errors from code review.

**Comment by marmarek** [comment]
PipelineRefresh

**Review by marmarek — APPROVED** [review]

**Pull request merged**